### PR TITLE
[COLLECTIONS-779] - Migrate assert to Assertions

### PR DESCRIPTION
**Pull request by arturobernalg**
commons-collections still needs a lot of work to complete the migration to Junit4.

### CODE REVIEW

File: src/test/java/org/apache/commons/collections4/iterators/LoopingListIteratorTest.java [hunk 1]
```diff
@@ -21,8 +21,8 @@
 import java.util.List;
 import java.util.NoSuchElementException;
 
-import org.junit.Test;
-import static org.junit.Assert.*;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/collections4/iterators/LoopingListIteratorTest.java:25]: @arturobernalg 
Did you forget to flip the arguments on the `assert*` API calls in this file and subsequent files?

Comment (arturobernalg) [reply src/test/java/org/apache/commons/collections4/iterators/LoopingListIteratorTest.java:25]: HI @garydgregory 

Nop. I don't.  I working on change/migrate the rest of the class. Trying to find out how migrate the BulkTest that extends TestCase hehehe. 

Any advice is more than welcome

Comment (garydgregory) [reply src/test/java/org/apache/commons/collections4/iterators/LoopingListIteratorTest.java:25]: I don't have specific advice, that is some older code initially based on JUnit 3.

Comment (arturobernalg) [reply src/test/java/org/apache/commons/collections4/iterators/LoopingListIteratorTest.java:25]: I can imagine. My idea it's split the migration in several PR.  In that way i can reduce the number of file to review and reduce the errors

File: src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionBuilderTest.java [hunk 2]
```diff
@@ -86,14 +89,14 @@ public void createPredicatedCollectionWithNotNullPredicate() {
...
+        assertEquals(1, collection.size());
 
         collection.add("test2");
-        Assert.assertEquals(2, collection.size());
+        assertEquals(2, collection.size());
 
         try {
             collection.add(null);
-            Assert.fail("Expecting IllegalArgumentException for failing predicate!");
+            fail("Expecting IllegalArgumentException for failing predicate!");
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionBuilderTest.java:99]: Use assertThrows()

File: src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionBuilderTest.java [hunk 3]
```diff
@@ -122,17 +125,17 @@ public void createPredicatedCollectionWithPredicate() {
     }
 
     private void checkPredicatedCollection2(final Collection<Integer> collection) {
-        Assert.assertEquals(2, collection.size());
+        assertEquals(2, collection.size());
 
         try {
             collection.add(4);
-            Assert.fail("Expecting IllegalArgumentException for failing predicate!");
+            fail("Expecting IllegalArgumentException for failing predicate!");
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionBuilderTest.java:132]: Use assertThrows()

**Review by garydgregory — CHANGES_REQUESTED** [review]

File: src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionBuilderTest.java (hunk 2 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionBuilderTest.java]: Done

File: src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionBuilderTest.java (hunk 3 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/collections4/collection/PredicatedCollectionBuilderTest.java]: Done

**Pull request merged**